src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/ApartmentValueFunction.java

```diff
@@ -147,7 +139,7 @@ public void setNbBathroomsValueFunction(PartialValueFunction<Double> nbBathrooms
    * @param terraceValueFunction
    */
   public void setTerraceValueFunction(PartialValueFunction<Boolean> terraceValueFunction) {
-    this.terraceValueFunction = checkNotNull(terraceValueFunction);
+    this.setInternalBooleanValueFunction(Criterion.TERRACE, checkNotNull(terraceValueFunction));
```

Comment:
Je ne suis pas sûre qu'il faille refaire un checkNotNull sur le deuxième paramètre car on vérifie déjà dans les fonctions setInternalBooleanValueFunction et setInternalDoubleValueFunction si ce paramètre est nul.
(Ici et dans tous les autres set)

Reply:
Fix in d34a55ad54fb6e348608d341481c740cbb7ff702